core/trino-main/src/main/java/io/trino/testing/MaterializedResult.java

```diff
@@ -430,7 +441,11 @@ public synchronized Builder queryDataEncoding(String encoding)
 
         public synchronized MaterializedResult build()
         {
-            return new MaterializedResult(rows.build(), types, columnNames, queryDataEncoding);
+            if ((session instanceof FullConnectorSession fullConnectorSession)) {
```

Comment:
what is the case when we are getting different session type here (TestingConnectorSession I presume)?
Please capture that as a comment.

Reply:
Yes, TestingConnectorSession

Reply:
Comment added. Will push later